src/ucp/proto/proto_common.c

```diff
@@ -146,27 +146,45 @@ static void ucp_proto_common_update_lane_perf_by_distance(
     perf->sys_latency += distance->latency;
 }
 
-void ucp_proto_common_get_lane_perf(const ucp_proto_common_init_params_t *params,
-                                    ucp_lane_index_t lane,
-                                    ucp_proto_common_tl_perf_t *perf)
+ucs_status_t
+ucp_proto_common_get_lane_perf(const ucp_proto_common_init_params_t *params,
+                               ucp_lane_index_t lane,
+                               ucp_proto_common_tl_perf_t *perf)
 {
-    const uct_iface_attr_t *iface_attr =
-            ucp_proto_common_get_iface_attr(&params->super, lane);
-    ucp_worker_h worker   = params->super.worker;
-    ucp_context_h context = worker->context;
+    ucp_worker_h worker        = params->super.worker;
+    ucp_context_h context      = worker->context;
+    ucp_rsc_index_t rsc_index  = ucp_proto_common_get_rsc_index(&params->super,
+                                                                lane);
+    ucp_worker_iface_t *wiface = ucp_worker_iface(worker, rsc_index);
```

Comment:
[minor] can it be `const`?

Reply:
in all other places, we don't use const qualifiers for objects like iface, worker, context, etc. and also ucp_worker_iface() returns a non-const pointer

Reply:
I see, thanks